java/system-test/system-test-suite/src/test/java/sleeper/systemtest/suite/testutil/FileInfoSystemTestHelper.java

```diff
@@ -16,30 +16,70 @@
 
 package sleeper.systemtest.suite.testutil;
 
+import sleeper.core.key.Key;
 import sleeper.core.partition.Partition;
+import sleeper.core.partition.PartitionTree;
 import sleeper.core.schema.Schema;
 import sleeper.core.statestore.FileInfo;
 import sleeper.core.statestore.FileInfoFactory;
 import sleeper.systemtest.suite.dsl.SleeperSystemTest;
 
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 
 public class FileInfoSystemTestHelper {
+    private final SleeperSystemTest sleeper;
 
-    private FileInfoSystemTestHelper() {
+    private FileInfoSystemTestHelper(SleeperSystemTest sleeper) {
+        this.sleeper = sleeper;
     }
 
-    public static FileInfoFactory fileInfoFactory(SleeperSystemTest sleeper) {
-        return new FileInfoFactory(sleeper.tableProperties().getSchema(), sleeper.partitioning().allPartitions());
+    public static FileInfoSystemTestHelper fileInfoHelper(SleeperSystemTest sleeper) {
+        return new FileInfoSystemTestHelper(sleeper);
     }
 
     public static FileInfoFactory fileInfoFactory(
             Schema schema, String tableId, Map<String, List<Partition>> allPartitionsByTable) {
-        return new FileInfoFactory(schema, allPartitionsByTable.get(tableId));
+        return FileInfoFactory.from(schema, allPartitionsByTable.get(tableId));
     }
 
     public static long numberOfRecordsIn(List<? extends FileInfo> files) {
         return files.stream().mapToLong(FileInfo::getNumberOfRecords).sum();
     }
+
+    private FileInfoFactory fileInfoFactory() {
+        return FileInfoFactory.from(sleeper.tableProperties().getSchema(),
+                sleeper.partitioning().allPartitions());
+    }
+
+    public FileInfo partitionFile(long records, Object min, Object max) {
+        return fileInfoFactory().partitionFile(getPartitionId(min, max), records);
```

Comment:
This will load all partitions every time this method is called. The old name leafFile also seems more specific.